cmd/meroxa/root/apps/deploy.go

```diff
@@ -189,13 +198,194 @@ func (d *Deploy) createApplication(ctx context.Context, pipelineUUID, gitSha str
 	return nil
 }
 
+// uploadSource creates first a Dockerfile to then, package the entire folder which will be later uploaded
+// this should ignore .git files and fixtures/
+func (d *Deploy) uploadSource(ctx context.Context, appPath, url string) error {
+	// Before creating a .tar.zip, we make sure it contains a Dockerfile
+	err := turbine.CreateDockerfile(appPath)
+	if err != nil {
+		return err
+	}
+
+	dFile := fmt.Sprintf("turbine-%s.tar.gz", d.appName)
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+	d.logger.Infof(ctx, "Packaging application located at %q...", appPath)
+	_ = turbineCLI.CreateTarAndZipFile(appPath, &buf)
+
+	err = os.Chdir(pwd)
+	if err != nil {
+		return err
+	}
+
+	fileToWrite, err := os.OpenFile(dFile, os.O_CREATE|os.O_RDWR, os.FileMode(0777))
+	defer func(fileToWrite *os.File) {
+		err := fileToWrite.Close()
+		if err != nil {
+			panic(err.Error())
+		}
+	}(fileToWrite)
+
+	if err != nil {
+		panic(err)
+	}
+	if _, err := io.Copy(fileToWrite, &buf); err != nil {
+		panic(err)
+	}
+
+	// We clean up Dockerfile as last step
+	err = os.Remove(filepath.Join(appPath, "Dockerfile"))
+	if err != nil {
+		return err
+	}
+
+	err = d.uploadFile(ctx, dFile, url)
+	if err != nil {
+		return err
+	}
+
+	// remove .tar.gz file
+	return os.Remove(dFile)
+}
+
+func (d *Deploy) uploadFile(ctx context.Context, filePath, url string) error {
+	d.logger.Info(ctx, "Uploading file to our build service...")
+	fh, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer func(fh *os.File) {
+		err := fh.Close()
+		if err != nil {
+
+		}
+	}(fh)
+
+	req, err := http.NewRequest("PUT", url, fh)
+	if err != nil {
+		return err
+	}
+
+	fi, err := fh.Stat()
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Accept", "*/*")
+	req.Header.Set("Content-Type", "multipart/form-data")
+	req.Header.Set("Accept-Encoding", "gzip, deflate, br")
+	req.Header.Set("Connection", "keep-alive")
+
+	req.ContentLength = fi.Size()
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		d.logger.Infof(ctx, "Uploaded!")
+		if err != nil {
+			d.logger.Error(ctx, err.Error())
+		}
+
+	}(res.Body)
+
+	return nil
+}
+
+func (d *Deploy) getPlatformImage(ctx context.Context, appPath string) (string, error) {
+	s, err := d.client.CreateSource(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	err = d.uploadSource(ctx, appPath, s.PutUrl)
+
+	sourceBlob := meroxa.SourceBlob{Url: s.GetUrl}
+	buildInput := &meroxa.CreateBuildInput{SourceBlob: sourceBlob}
+
+	build, err := d.client.CreateBuild(ctx, buildInput)
+	if err != nil {
+		return "", err
+	}
+
+	fmt.Print("Getting status for build: ", build.Uuid)
+	for {
```

Comment:
seems dangerous to have no timeout, but i guess it's easy enough to shutdown the CLI

Reply:
@janelletavares that's right, you could still Ctrl+C if you have to. I could still add a time out though